### PR TITLE
bwl: dwpal: fix sta_wlan_hal_dwpal::add_network return code

### PR DESCRIPTION
**Pull request by mariomaz**
Return code of method sta_wlan_hal_dwpal::add_network() must be -1 on error.

### CODE REVIEW

**Review by VladyslavTupikin — APPROVED** [review]

**Review by adam1985d — APPROVED** [review]
please provide a description of platform-tests done to verify this change.
how did it work until now?

File: common/beerocks/bwl/dwpal/sta_wlan_hal_dwpal.cpp [hunk 1]
```diff
@@ -558,7 +558,7 @@ int sta_wlan_hal_dwpal::add_network()
     // Send command
     if (!dwpal_send_cmd(cmd, &reply)) {
         LOG(ERROR) << "ADD_NETWORK failed!";
-        return false;
+        return -1;
```
Comment (adam1985d) [inline common/beerocks/bwl/dwpal/sta_wlan_hal_dwpal.cpp]:
```suggestion
        return INVALID_NETWORK_ID;
```
and define a `constexpr` in the appropriate place

**Review by adam1985d — APPROVED** [review]
one small comment - approving since I think the fix is simple and doesn't require re-review.

File: common/beerocks/bwl/dwpal/sta_wlan_hal_dwpal.cpp (hunk 1 above)
Comment (adam1985d) [reply common/beerocks/bwl/dwpal/sta_wlan_hal_dwpal.cpp]: and change `sta_wlan_hal_dwpal::connect` (point of use) accordingly.

**Comment by mariomaz** [comment]
> please provide a description of platform-tests done to verify this change.
> how did it work until now?

The reason no test has been failing is probably that there was no test checking this particular error condition. AFAIK, test coverage is very low and limited to the happy path. This was a very good candidate of error to be spotted with unit tests.

**Pull request merged**